.github/workflows/ci.yaml

```diff
@@ -0,0 +1,131 @@
+        required: true
+        default: angular-react-starter-image
+        type: string
+      docker_tag:
+        description: Docker tag
+        required: true
+        default: latest
+        type: string
+      mail_server_address:
+        description: Mail server address
+        required: true
+        default: smtp.gmail.com
+        type: string
+      mail_server_port:
+        description: Mail server port
+        required: true
+        default: 465
+        type: number
+      mail_receiver: 
+        description: Email of the receiver
+        required: false
+        type: string
+      mail_subject:
+        description: Subject of the email
+        required: true
+        default: Github actions job result
+        type: string
+    secrets:
+      mail_username:
+        description: Username for the email notification step
+        required: true
+      mail_password:
+        description: Password for the email notification step
+        required: true
+      github_token:
+        description: Github token for accessing the github container registry
+        required: true
+
+env:
+  IMAGE_NAME: '${{ inputs.docker_registry }}/${{ github.actor }}/${{ inputs.docker_repo }}:${{ inputs.docker_tag }}'
+  DEFAULT_MAIL_RECEIVER: '${{ github.actor_id }}+${{ github.actor }}@users.noreply.github.com'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./angular
+    steps:
+      - name: Checkout code from the repository
+        uses: actions/checkout@v4
+
+      - name: Set up node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ inputs.node_version }}
+
+      - name: Install clean dependencies
+        run: npm ci
+
+      - name: Run headless tests
+        id: run-tests
+        run: npm run test -- --watch=false --browsers=ChromeHeadless 
+
+      - name: Build angular project
+        id: build-project
+        run: npm run build
+
+      - name: Log in to github container registry  
+        id: login-registry
+        run: | 
+          echo "${{ secrets.github_token }}" \
+          | docker login ${{ inputs.docker_registry }} -u ${{ github.actor }} --password-stdin
+
+      - name: Build docker image
+        id: build-docker
+        run: docker build -t $IMAGE_NAME .
+        working-directory: .
+
+      - name: Push docker image
+        id: push-docker
+        run: docker push $IMAGE_NAME
+
+      - name: Get pipeline execution time
+        id: get-time
+        if: always()
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          run_info=$(gh run list --json createdAt,updatedAt -w ci.yaml -L 1)
+          start_time=$(echo "$run_info" | jq -r '.[0].createdAt')
+          end_time=$(echo "$run_info" | jq -r '.[0].updatedAt')
+          duration=$(($(date -d "$end_time" +%s) - $(date -d "$start_time" +%s)))
+          echo "execution_time=${duration}s" >> $GITHUB_ENV
```

Comment:
let's simplify this and make it oneliner.

Reply:
Done